pkg/gadget-collection/gadgets/top/ebpf/gadget.go

```diff
@@ -0,0 +1,336 @@
+// Copyright 2019-2022 The Inspektor Gadget authors
```

Comment:
We are trying to make the tracers usable by 3rd party applications: https://github.com/kinvolk/inspektor-gadget/issues/723. I think most of the logic on this file should be moved to `pkg/gadgets/top/ebpf/tracer.go` and this file should only contain logic related to the trace resource. (I think you can take inspiration from top file, that uses a callback to pass events from tracer.go to gadget.go).

Reply:
I moved the logic and implemented similarly to top file like you suggested.